static/app/views/stories/storyTypes.tsx

```diff
@@ -0,0 +1,432 @@
+import {Fragment, useCallback, useMemo, useState} from 'react';
+import type {PropItem, Props} from 'react-docgen-typescript';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import {InputGroup} from 'sentry/components/inputGroup';
+import {Tooltip} from 'sentry/components/tooltip';
+import {IconChevron} from 'sentry/icons';
+import {IconSearch} from 'sentry/icons/iconSearch';
+import {space} from 'sentry/styles/space';
+import {fzf} from 'sentry/utils/profiling/fzf/fzf';
+
+interface StoryTypesProps {
+  types: TypeLoader.ComponentDocWithFilename | undefined;
+}
+
+export function StoryTypes(props: StoryTypesProps) {
+  const [query, setQuery] = useState('');
+  const nodes = usePropTree(props.types?.props ?? {}, query);
+
+  const onSearchInputChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setQuery(e.target.value);
+    },
+    [setQuery]
+  );
```

Comment:
is there a reason we’re memoizing this function? As far as I can see, `InputGroup.Input` is not a memoized component so I don’t think this does much.

we could also just inline it on the call-side:

```
onChange={(e) => setQuery(e.target.value)}
```

Reply:
Oh yeah, I had this syncing with query param and ultimately removed it as it felt unnecessary